xilinx/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+pub mod sys;
+pub use sys::*;
+
+/*---- decoder ----*/
+pub mod xlnx_decoder;
+pub use xlnx_decoder::*;
+
+pub mod xlnx_dec_utils;
+pub use xlnx_dec_utils::*;
+
+pub mod xlnx_dec_props;
+pub use xlnx_dec_props::*;
+
+/*---- error ----*/
+pub mod xlnx_error;
+pub use xlnx_error::*;
+
+const XCLBIN_FILENAME: &[u8] = b"/opt/xilinx/xcdr/xclbins/transcode.xclbin\0";
+
+/// initalized the number of devices specified with the default xclbin_name
+///
+/// @decice_count: The number of devices available in the system.
+pub fn xlnx_init_all_devices(device_count: i32) -> Result<(), simple_error::SimpleError> {
+    let mut xclbin_params = Vec::new();
+
+    for i in 0..device_count {
+        xclbin_params.push(XmaXclbinParameter {
+            xclbin_name: XCLBIN_FILENAME.as_ptr() as *mut i8,
+            device_id: i,
+        });
+    }
+
+    let ret = unsafe { xma_initialize(xclbin_params.as_mut_ptr(), device_count) };
+    if ret as u32 != XMA_SUCCESS {
+        simple_error::bail!("Xma initalization failed")
+    }
+
+    Ok(())
+}
+
+pub fn xrm_precision_1000000_bitmask(val: i32) -> i32 {
+    val << 8
+}
+
+// performs a strcpy from string literal to existing output array. ensures null termination.
+pub fn strcpy_to_arr_i8(buf: &mut [i8], in_str: &str) -> Result<(), simple_error::SimpleError> {
```

Comment:
This and several other things here probably shouldn't be public. `pub(crate)` maybe:

```suggestion
pub(crate) fn strcpy_to_arr_i8(buf: &mut [i8], in_str: &str) -> Result<(), simple_error::SimpleError> {
```